uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/FederatedLoginIT.java

```diff
@@ -0,0 +1,168 @@
+package org.cloudfoundry.identity.uaa.integration.feature;
+
+import org.cloudfoundry.identity.uaa.ServerRunning;
+import org.cloudfoundry.identity.uaa.integration.feature.orchestrator.uilocators.IdploginUI;
+import org.cloudfoundry.identity.uaa.integration.feature.orchestrator.uilocators.SploginUI;
+import org.cloudfoundry.identity.uaa.integration.feature.orchestrator.utils.IntegrationUtilsOrchestrator;
+import org.cloudfoundry.identity.uaa.integration.util.IntegrationTestUtils;
+import org.cloudfoundry.identity.uaa.provider.IdentityProvider;
+import org.cloudfoundry.identity.uaa.provider.SamlIdentityProviderDefinition;
+import org.cloudfoundry.identity.uaa.provider.saml.idp.SamlServiceProvider;
+import org.cloudfoundry.identity.uaa.provider.saml.idp.SamlServiceProviderDefinition;
+import org.cloudfoundry.identity.uaa.provider.saml.idp.SamlTestUtils;
+import org.cloudfoundry.identity.uaa.scim.ScimGroup;
+import org.cloudfoundry.identity.uaa.zone.IdentityZone;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.oauth2.common.util.RandomValueStringGenerator;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+import java.net.Inet4Address;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import static org.junit.Assert.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = DefaultIntegrationTestConfig.class)
+public class FederatedLoginIT extends BaseSamlTestIT {
```

Comment:
Naming of this test still needs some improvement so it's clear that we are testing federated login for orchestrator

I suggest two options. Please choose one:

Option 1: Move FederatedLoginIT to org.cloudfoundry.identity.uaa.integration.feature.orchestrator package. That will clearly separate this test from existing IT test suite. If you choose this option, make sure it satisfies exclude criteria for MockMvc and include criteria for integrationTests tasks.

Option 2: Rename FederatedLoginIT to SamlLoginForOrchestratorZoneIT and BaseSamlTestIT to SamlBaseIT. I added Saml prefix so all Saml tests are grouped together in org.cloudfoundry.identity.uaa.integration.feature package